src/helm/benchmark/scenarios/medalign_scenario.py

```diff
@@ -80,61 +138,462 @@ def get_ehrs(path_to_ehrs: str) -> Dict[int, str]:
+    Parameters:
+        ehr_visit_strs (list of str): List of EHR visit strings.
+        query (str): Query string to retrieve relevant visits.
+        target_length (int): Maximum total token count for the final list of documents.
+        tokenizer (Callable): Tokenizer that converts text to tokens (used for tracking context length)
 
+    Returns:
+        list[str]: List of EHR visit strings sorted chronologically and constrained by the target length.
+    """
+    langchain_docs = [langchain.schema.Document(page_content=doc) for doc in ehr_visit_strs]
+
+    # `k` is the number of documents to retrieve
+    # We retrieve everything and just use the BM25Retriever to sort the documents
+    retriever = BM25Retriever.from_documents(langchain_docs, k=len(langchain_docs))
+
+    # Invoking the retriever means the most relevant documents are sorted first
+    sorted_docs = retriever.invoke(query)
+
+    # Define the regex pattern to find the start time
+    # pattern = r'start="([\d/]+ [\d:]+)"'
+    pattern = r'start="([\d/]+ [\d:]+ ?[APM]{0,2})"'
+
+    docs = []
+    dts = []
+
+    # Find the startime of the document
+    for doc in sorted_docs:
+        doc_content = doc.page_content
+        start_dt_match = re.search(pattern, doc_content)
+        if start_dt_match:
+            start_dt = start_dt_match.group(1)
+            parsed = False
+            # Try different date formats
+            for fmt in (
+                "%m/%d/%y %I:%M %p",
+                "%m/%d/%Y %I:%M %p",
+                "%m/%d/%y %H:%M",
+                "%m/%d/%Y %H:%M",
+            ):
+                try:
+                    dts.append(datetime.datetime.strptime(start_dt, fmt))
+                    parsed = True
+                    break
+                except ValueError:
+                    continue
+            if not parsed:
+                print(f"Error parsing date: {start_dt}")
+                continue
+        else:
+            print("Start time not found.")
+            dts.append(datetime.datetime.min)
+        docs.append(doc_content)
+
+    final_docs = []
+    current_length = 0
+
+    # Add documents until we exceed the allocated context length
+    for i in range(len(docs)):
+        doc_content = docs[i]
+        doc_length = len(tokenizer.encode(doc_content))
+        final_docs.append((dts[i], doc_content))
+        current_length += doc_length
+        if current_length > target_length:
+            break
+
+    # Sort final_docs chronologically
+    final_docs.sort(key=lambda x: x[0])
+
+    # Extract only the document content for the final output
+    final_docs_content = [doc_content for _, doc_content in final_docs]
+
+    return final_docs_content
+
+
+def get_prompt_template(path_to_template="prompt_templates/generic.txt"):
+    with open(path_to_template, encoding="utf-8", mode="r") as f:
+        prompt_template_str = f.read()
+    prompt_template_obj = langchain.prompts.PromptTemplate.from_template(prompt_template_str)
```

Comment:
Minor, optional: I would prefer to do this with raw string templates, but this might be too late to change now.